framework/src/utils/FunctionMaterialPropertyDescriptor.C

```diff
@@ -129,40 +138,54 @@ FunctionMaterialPropertyDescriptor<is_ad>::parseDerivative(const std::string & e
 
     return;
   }
-  else if (open != std::string::npos && close != std::string::npos &&
-           expression.substr(0, open) == "D")
+  if (open != std::string::npos && close != std::string::npos)
   {
-    // tokenize splits the arguments in d2h2:=D[h2(eta1,eta2),eta1] into 'h2(eta1' 'eta2)' 'eta1'
-    // DAMN!!
-    auto arguments = expression.substr(open + 1, close - open - 1);
-    auto close2 = arguments.find_last_of(")");
-
-    if (close2 == std::string::npos)
+    if (expression.substr(0, open) == "Old")
+    {
+      _base_name = expression.substr(open + 1, close - open - 1);
+      _dependent_symbols.clear();
+      _state = PropertyState::OLD;
+      return;
+    }
+    if (expression.substr(0, open) == "Older")
     {
-      // rest of argument list 0 is the function and 1,.. are the variable to take the derivative
-      // w.r.t.
-      MooseUtils::tokenize(arguments, _derivative_symbols, 0, ",");
+      _base_name = expression.substr(open + 1, close - open - 1);
+      _dependent_symbols.clear();
+      _state = PropertyState::OLDER;
+      return;
+    }
+    else if (expression.substr(0, open) == "D")
```

Comment:
should you set the state here?
you have one constructor that does not default it